### PR TITLE
Add docker setup

### PR DESCRIPTION
**Pull request by b-admike**
### Description
Add a script which downloads an Oracle 11g docker image and sets up a container from it.

#### Related issues
connect to https://github.com/strongloop-internal/scrum-apex/issues/216

### Checklist

- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -469,9 +469,25 @@ Destroying models may result in errors due to foreign key integrity. First delet
...
-to run them using our preconfigured test server.
+### Own instance
+If you have a local or remote Oracle instance and would like to use that to run the test suite, use the following command:
+- Linux
+```bash
+ORACLE_HOST=<HOST> ORACLE_PORT=<PORT> ORACLE_USER=<USER> ORACLE_PASSWORD=<PASSWORD> ORACLE_DATABASE=<DATABASE> npm test
+```
+- Windows
+```bash
+SET ORACLE_HOST=<HOST> SET ORACLE_PORT=<PORT> SET ORACLE_USER=<USER> SET ORACLE_PASSWORD=<PASSWORD> SET ORACLE_DATABASE=<DATABASE> npm test
```
Comment (dhmlau) [inline README.md]: i believe those are separate commands rather than one liner.

Comment (ssh24) [inline README.md]: I think they work in one line, no?

File: README.md [hunk 2]
```diff
@@ -469,9 +469,25 @@ Destroying models may result in errors due to foreign key integrity. First delet
...
-1. Ask a core developer for instructions on how to set up test server
-   credentials on your machine
-2. `npm test`
+### Docker
+If you do not have a local Oracle instance, you can also run the test suite with very minimal requirements.
+- Assuming you have [Docker](https://docs.docker.com/engine/installation/) installed, run the following script which would spawn an Oracle instance on your local machine:
+```bash
+source setup.sh <HOST> <PORT>
+```
+where `<HOST>`, and `<PORT>` are optional parameters. The default values are `localhost`, and `1521`. The `<USER>`, `PASSWORD`, and `DATABASE` settings are always `system`, `oracle`, and `XE` respectively.
```
Comment (ssh24) [inline README.md]: @b-admike Any specific reason we are not allowing the user to input their own `user` and `password`?

File: setup.sh [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+## run the oracle container
+printf "\n${RED}>> Starting the oracle container${PLAIN} ${GREEN}...${PLAIN}"
+docker run --name $ORACLE_CONTAINER -p $PORT:1521 -d sath89/oracle-xe-11g:latest 2>&1
+
+##wait for orale database container to be ready
+OUTPUT=1
+TIMEOUT=300
+TIME_PASSED=0
+WAIT_STRING="+"
```
Comment (ssh24) [inline setup.sh]: can we set the `WAIT_STRING` to be `.` instead?

File: setup.sh [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+printf "\n${CYAN}Clean up complete.${PLAIN}\n"
+
+## pull latest oracle image
+printf "\n${RED}>> Pulling latest oracle image${PLAIN} ${GREEN}...${PLAIN}"
+docker pull sath89/oracle-xe-11g:latest > /dev/null 2>&1
+printf "\n${CYAN}Image successfully built.${PLAIN}\n"
+
+## run the oracle container
+printf "\n${RED}>> Starting the oracle container${PLAIN} ${GREEN}...${PLAIN}"
+docker run --name $ORACLE_CONTAINER -p $PORT:1521 -d sath89/oracle-xe-11g:latest 2>&1
```
Comment (ssh24) [inline setup.sh]: use `> /dev/null 2>&1` instead of `2>&1`

File: setup.sh [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+##wait for orale database container to be ready
+OUTPUT=1
+TIMEOUT=300
+TIME_PASSED=0
+WAIT_STRING="+"
+START_MESSAGE="Database ready to use."
+printf "${RED}Waiting for database to be ready${PLAIN} ${GREEN}...${PLAIN}"
```
Comment (ssh24) [inline setup.sh]: Add a new line at the front and end of this:
`printf "\n${RED}Waiting for database to be ready${PLAIN} ${GREEN}...${PLAIN}\n"`

File: README.md (hunk 1 above)
Comment (dhmlau) [inline README.md]: nope. I just tried it on my Win7 machine. 
```
C:\Users\Administrator>set ABC=a SET DEF=d

C:\Users\Administrator>echo %ABC%
a SET DEF=d

C:\Users\Administrator>set ABC=a; SET DEF=d

C:\Users\Administrator>echo %ABC%
a; SET DEF=d
```

Comment (dhmlau) [inline README.md]: @ssh24  actually i put a similar comment on ur PR too.

Comment (ssh24) [inline README.md]: Ah. Will update the rest of the connectors.

File: README.md (hunk 2 above)
Comment (b-admike) [inline README.md]: Good point. We could and I will add it.

File: setup.sh [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+## color codes
+RED='\033[1;31m'
+GREEN='\033[1;32m'
+YELLOW='\033[1;33m'
+CYAN='\033[1;36m'
+PLAIN='\033[0m'
+
+## variables
+ORACLE_CONTAINER="oracle_c"
```
Comment (cjbj) [inline setup.sh]: If the access and licensing and size works for you, there is also a 12.2 DB you could use: https://store.docker.com/images/oracle-database-enterprise-edition

Comment (b-admike) [inline setup.sh]: Thank you. I think for now the image we use is sufficient, but in the future we can look into it.

**Comment by b-admike** [comment]
@slnode test please

**Pull request merged**